packages/nimbus/src/components/time-input/time-input.dev.mdx

```diff
@@ -0,0 +1,724 @@
+        onChange={setTime}
+        minValue={minTime}
+        maxValue={maxTime}
+        isInvalid={isInvalid}
+      />
+      <Text fontSize="sm" color={isInvalid ? "critical.11" : "neutral.11"}>
+        {isInvalid 
+          ? "Time must be between 9:00 AM and 5:00 PM"
+          : "Valid time range: 9:00 AM - 5:00 PM"}
+      </Text>
+    </Stack>
+  );
+}
+```
+
+### Disabled state
+
+Prevent user interaction with the `isDisabled` prop:
+
+```jsx-live-dev
+const App = () => (
+  <TimeInput
+    isDisabled
+    aria-label="Disabled time input"
+    defaultValue={new Time(10, 30)}
+  />
+)
+```
+
+### Invalid state
+
+Indicate validation errors with the `isInvalid` prop:
+
+```jsx-live-dev
+const App = () => (
+  <Stack direction="column" gap="400">
+    <TimeInput
+      isInvalid
+      aria-label="Invalid time"
+      defaultValue={new Time(8, 0)}
+    />
+    <Text fontSize="sm" color="critical.11">
+      Selected time is outside business hours
+    </Text>
+  </Stack>
+)
+```
+
+### Read-only state
+
+Display time values without allowing edits:
+
+```jsx-live-dev
+const App = () => (
+  <TimeInput
+    isReadOnly
+    aria-label="Read-only time"
+    defaultValue={new Time(14, 30)}
+  />
+)
+```
+
+### Required state
+
+Mark the input as required for form validation:
+
+```jsx-live-dev
+const App = () => (
+  <TimeInput
+    isRequired
+    aria-label="Required time input"
+  />
+)
+```
+
+### Uncontrolled mode
+
+For simpler use cases, use uncontrolled mode with `defaultValue` and `onChange`:
+
+```jsx-live-dev
+const App = () => {
+  const [displayValue, setDisplayValue] = useState<string>('No time selected');
+
+  return (
+    <Stack direction="column" gap="400">
+      <TimeInput
+        defaultValue={new Time(9, 30)}
+        onChange={(value) => {
+          if (value) {
+            setDisplayValue(`Selected: ${value.toString()}`);
+          } else {
+            setDisplayValue('No time selected');
+          }
+        }}
+        aria-label="Uncontrolled time input"
+      />
+      <Text fontSize="sm">{displayValue}</Text>
+    </Stack>
+  );
+}
+```
+
+Use uncontrolled mode when you need to capture the selected time without managing state yourself.
+
+### Controlled mode
+
+For scenarios requiring programmatic control or coordination with other components, use controlled mode:
+
+```jsx-live-dev
+const App = () => {
+  const [value, setValue] = useState<TimeInputProps["value"]>(new Time(12, 0));
+
+  return (
+    <Stack direction="column" gap="400">
+      <TimeInput
+        value={value}
+        onChange={setValue}
+        aria-label="Controlled time input"
+      />
+      <Stack direction="row" gap="300">
+        <Button size="sm" onPress={() => setValue(new Time(9, 0))}>
+          Set to 9:00 AM
+        </Button>
+        <Button size="sm" onPress={() => setValue(new Time(17, 0))}>
+          Set to 5:00 PM
+        </Button>
+        <Button size="sm" variant="secondary" onPress={() => setValue(null)}>
```

Comment:
Once again, AI out here slapping us with that `sm` Button prop. Let's just remove it so we have the default Button.